xds/src/main/java/io/grpc/xds/EnvoyServerProtoData.java

```diff
@@ -201,29 +218,62 @@ public String toString() {
     private final int destinationPort;
     private final List<CidrRange> prefixRanges;
     private final List<String> applicationProtocols;
+    private final List<CidrRange> sourcePrefixRanges;
+    private final ConnectionSourceType sourceType;
+    private final List<Integer> sourcePorts;
 
     @VisibleForTesting
-    FilterChainMatch(int destinationPort,
-        List<CidrRange> prefixRanges, List<String> applicationProtocols) {
+    FilterChainMatch(
+        int destinationPort,
+        List<CidrRange> prefixRanges,
+        List<String> applicationProtocols,
+        List<CidrRange> sourcePrefixRanges,
+        ConnectionSourceType sourceType,
+        List<Integer> sourcePorts) {
       this.destinationPort = destinationPort;
       this.prefixRanges = Collections.unmodifiableList(prefixRanges);
       this.applicationProtocols = Collections.unmodifiableList(applicationProtocols);
+      this.sourcePrefixRanges = sourcePrefixRanges;
+      this.sourceType = sourceType;
+      this.sourcePorts = sourcePorts;
     }
 
     static FilterChainMatch fromEnvoyProtoFilterChainMatch(
-        io.envoyproxy.envoy.config.listener.v3.FilterChainMatch proto) {
+        io.envoyproxy.envoy.config.listener.v3.FilterChainMatch proto)
+        throws InvalidProtocolBufferException {
       List<CidrRange> prefixRanges = new ArrayList<>();
       for (io.envoyproxy.envoy.config.core.v3.CidrRange range : proto.getPrefixRangesList()) {
         prefixRanges.add(CidrRange.fromEnvoyProtoCidrRange(range));
       }
+      List<CidrRange> sourcePrefixRanges = new ArrayList<>();
+      for (io.envoyproxy.envoy.config.core.v3.CidrRange range : proto.getSourcePrefixRangesList()) {
+        sourcePrefixRanges.add(CidrRange.fromEnvoyProtoCidrRange(range));
+      }
       List<String> applicationProtocols = new ArrayList<>();
-      for (String appProtocol  : proto.getApplicationProtocolsList()) {
+      for (String appProtocol : proto.getApplicationProtocolsList()) {
         applicationProtocols.add(appProtocol);
       }
+      ConnectionSourceType sourceType = null;
+      switch (proto.getSourceType()) {
+        case ANY:
+          sourceType = ConnectionSourceType.ANY;
+          break;
+        case EXTERNAL:
+          sourceType = ConnectionSourceType.EXTERNAL;
+          break;
+        case SAME_IP_OR_LOOPBACK:
+          sourceType = ConnectionSourceType.SAME_IP_OR_LOOPBACK;
+          break;
+        default:
+          throw new InvalidProtocolBufferException("Unknown source-type:" + proto.getSourceType());
```

Comment:
This can be noop.

Reply:
[This says](https://github.com/envoyproxy/envoy/blob/main/api/envoy/config/listener/v3/listener_components.proto#L126) "Can be any, local or external network." . Also the validate_rules for this field says "defined_only" which means "defined_only: the field must be one of the specified values in the enum descriptor.". So `InvalidProtocolBufferException` quite appropriate here.

Reply:
Ok, I see there is also an enum value `UNRECOGNIZED` that is not run through. That's fair.